pkgs/applications/virtualization/xen/generic.nix

```diff
@@ -1,265 +1,696 @@
-config:
-{ lib, stdenv, cmake, pkg-config, which
-
-# Xen
-, bison, bzip2, checkpolicy, dev86, figlet, flex, gettext, glib
-, acpica-tools, libaio, libiconv, libuuid, ncurses, openssl, perl
-, xz, yajl, zlib
-, python3Packages
-
-# Xen Optional
-, ocamlPackages
-
-# Scripts
-, coreutils, gawk, gnused, gnugrep, diffutils, multipath-tools
-, iproute2, inetutils, iptables, bridge-utils, openvswitch, nbd, drbd
-, util-linux, procps, systemd
-
-# Documentation
-# python3Packages.markdown
-, fig2dev, ghostscript, texinfo, pandoc
-
-, binutils-unwrapped
-
-, ...} @ args:
-
-with lib;
+versionDefinition:
+{
+  lib,
+  stdenv,
+  autoPatchelfHook,
+  cmake,
+  ninja,
+  pkg-config,
+  testers,
+  which,
+
+  fetchgit,
+  fetchFromGitHub,
+
+  # Xen
+  acpica-tools,
+  bison,
+  bzip2,
+  dev86,
+  e2fsprogs,
+  flex,
+  gettext,
+  glib,
+  libaio,
+  libiconv,
+  libnl,
+  libuuid,
+  lzo,
+  ncurses,
+  ocamlPackages,
+  openssl,
+  perl,
+  python311Packages,
+  xz,
+  yajl,
+  zlib,
+  zstd,
```

Comment:
Having just spend a lot of time stripping bogus/obsolete deps out of Xen's CI build containers, here are a few I prepared earlier...

https://github.com/xen-project/xen/commit/1ddc8c1bad93aa6cbfe616dd72333460c47f96c9
https://github.com/xen-project/xen/commit/a8bc31be0103475772e23fc47fedcd09d71fe7c5

Obviously the package names will be a little different, but I have split them as best as I can by build area.  Runtime dependencies are a bit more tricky.

Pretty much any dockerfile in https://github.com/xen-project/xen/tree/master/automation/build which uses HEREDOCS (i.e. I patched it recently) should be stripped down to the actual dependencies, rather than an 2-decade-old list of stale ones.

Reply:
Thanks! Removed unused dependencies.